internal/testcase/ecs_pod.go

```diff
@@ -1 +1,159 @@
+		SetMemoryMB(128).
+		SetCPU(128).
+		SetTaskRole(testutil.TaskRole()).
+		AddTags("tag").
+		SetExecutionOptions(*execOpts)
+
+	optsSecret := cocoa.NewECSPodCreationOptions().
+		AddContainerDefinitions(*containerDef.AddEnvironmentVariables(*secret, *secretOwned)).
+		SetMemoryMB(128).
+		SetCPU(128).
+		SetTaskRole(testutil.TaskRole()).
+		AddTags("tag").
+		SetExecutionOptions(*execOpts)
+
+	return map[string]ECSPodTestCase{
+		"StopSucceeds": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, opts)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			err = p.Stop(ctx)
+			require.NoError(t, err)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+
+			assert.Equal(t, cocoa.Stopped, info.Status)
+		},
+		"StopSucceedsWithSecrets": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, optsSecret)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			require.NotNil(t, info.Resources)
+			assert.Equal(t, cocoa.Running, info.Status)
+			assert.Equal(t, 2, len(info.Resources.Secrets))
+
+			err = p.Stop(ctx)
+			require.NoError(t, err)
+
+			info, err = p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Stopped, info.Status)
+			require.Equal(t, 2, len(info.Resources.Secrets))
+
+			arn := info.Resources.Secrets[0].Name
+			id, err := v.GetValue(ctx, *arn)
+			require.NoError(t, err)
+			require.NotNil(t, id)
+		},
+		"StopFailsOnIncorrectPodStatus": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, opts)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			err = p.Stop(ctx)
+			require.NoError(t, err)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotZero(t, info)
+			assert.Equal(t, cocoa.Stopped, info.Status)
+
+			err = p.Stop(ctx)
+			require.Error(t, err)
+		},
+		"DeleteSucceeds": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, opts)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Running, info.Status)
+
+			err = p.Delete(ctx)
+			require.NoError(t, err)
+
+			info, err = p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Deleted, info.Status)
+		},
+		"DeleteSucceedsWithSecrets": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, optsSecret)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			require.NotNil(t, info.Resources)
+			require.NotNil(t, info.Resources.Secrets)
```

Comment:
You probably want to check `require.NotEmpty(t, info.Resources.Secrets)` because the slice could be empty. However, I would instead just remove this check since L137 already checks that the secrets are both non-empty and of length 2.